### ISSUE TITLE
Matplotlib RuntimeError() whenever Eureka is imported and plt.show() is called. 

### ISSUE THREAD

**Issue by AarynnCarter** [kevin218/Eureka#42]
Whenever Eureka is imported into a file, attempting to call plt.show() for a matplotlib.pyplot plot results in the error copied below. 

If Eureka is simply not imported, but the same anaconda environment is used, no such error appears. It's important to note that you don't have to actually use Eureka / specify any latex for this to appear. Simply importing the package whilst trying to call plt.show() results in this error. 

Temporary workaround is to add the following lines at some point in the code prior to any calls to plt.show()
```
from matplotlib import rc
rc('text', usetex=False)
```

However, this isn't the best long term solution. Current thought is that the error is being raised by the use of latex in label text within the Eureka package, and whether 'r' should be placed before the label text itself. I expect we need to discuss whether the 'r' is important or not, personally I've never had to include it and any latex commands (e.g. $\mu$) have been interpreted correctly by matplotlib. 

Final note, it seems this issue isn't unique to me, and at least 3 other people experienced this during the ERS hackathon. 

-------------------------------------------------------------------------------------------------------------------------

Mac OS Catalina 10.15.7
Python 3.9.5 Anaconda Environment
```
Traceback (most recent call last):
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/texmanager.py", line 252, in _run_checked_subprocess
    report = subprocess.check_output(
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'latex'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/backends/backend_macosx.py", line 45, in _draw
    self.figure.draw(renderer)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/artist.py", line 74, in draw_wrapper
    result = draw(artist, renderer, *args, **kwargs)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/artist.py", line 51, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/figure.py", line 2780, in draw
    mimage._draw_list_compositing_images(
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/image.py", line 132, in _draw_list_compositing_images
    a.draw(renderer)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/artist.py", line 51, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/_api/deprecation.py", line 431, in wrapper
    return func(*inner_args, **inner_kwargs)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/axes/_base.py", line 2921, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/image.py", line 132, in _draw_list_compositing_images
    a.draw(renderer)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/artist.py", line 51, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/axis.py", line 1137, in draw
    ticklabelBoxes, ticklabelBoxes2 = self._get_tick_bboxes(ticks_to_draw,
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/axis.py", line 1063, in _get_tick_bboxes
    return ([tick.label1.get_window_extent(renderer)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/axis.py", line 1063, in <listcomp>
    return ([tick.label1.get_window_extent(renderer)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/text.py", line 903, in get_window_extent
    bbox, info, descent = self._get_layout(self._renderer)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/text.py", line 306, in _get_layout
    _, lp_h, lp_d = renderer.get_text_width_height_descent(
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/backends/backend_agg.py", line 229, in get_text_width_height_descent
    w, h, d = texmanager.get_text_width_height_descent(
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/texmanager.py", line 399, in get_text_width_height_descent
    dvifile = self.make_dvi(tex, fontsize)
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/texmanager.py", line 291, in make_dvi
    self._run_checked_subprocess(
  File "/Users/acarter/anaconda3/envs/eureka/lib/python3.9/site-packages/matplotlib/texmanager.py", line 256, in _run_checked_subprocess
    raise RuntimeError(
RuntimeError: Failed to process string with tex because latex could not be found

```

**Comment by kevin218** [comment]
One workaround may be to run the above two lines only if an exception is thrown.  That way those with Latex can still enjoy the nice labels.  Long term, I'm not sure what a good solution would be.

**Comment by AarynnCarter** [comment]
I suppose the issue is that the user has to run those lines on their end, even if they don't directly use anything from Eureka. I'm not sure if it's possible to get the package to detect if a plt.show() causes a crash somewhere in an exterior file. 

To be clear though, I have Latex installed and want to enjoy nice labels too! I've attached a nonsense plot below which was made in the exact same anaconda environment as above, but without importing Eureka, as you can see the latex commands have been readily understood for the axis labels. 

```
import matplotlib.pyplot as plt

y= [1,2,3,4,5]
z=[10,11,12,13,15]

plt.plot(y,z)
plt.gca().set_ylabel(r'$\Sigma \mu$', fontsize=14)
plt.gca().set_xlabel(r'$\rho \pi$', fontsize=14)
plt.show()
```

![Figure_1](https://user-images.githubusercontent.com/23636747/123485970-089ee400-d5c0-11eb-939a-38dbda89f121.png)

**Comment by kevin218** [comment]
Can you tried the following suggestions from Stack Overflow?
https://stackoverflow.com/questions/54477891/filenotfounderror-errno-2-no-such-file-or-directory-latex-latex-python 
Maybe this will help us to identify a solution that we can then post in an FAQ section on the Eureka! webpage.

**Comment by sebastian-zieba** [comment]
When I ran Eureka! for the first time on a different machine, I got the same matplotlib error. 
I fixed it by checking out the same Stack Overflow Kevin just posted and installing: _sudo apt install cm-super_
I actually mentioned it in the docs quickstart (https://eurekadocs.readthedocs.io/en/latest/quickstart.html#run-eureka)

**Comment by AarynnCarter** [comment]
Okay, thanks very much for the links! Unfortunately I couldn't get Sebastian's solution to work, however from Kevin's link I was able to piece together a permanent solution... I think. 

The trick is to identify where your TeX installation is and manually add it to PATH in your bashrc or bash_profile (whichever you use). So, for my case the following line in my ~/.bashrc: 

`export PATH="~/anaconda3/bin:$PATH"`

Changes to...

`export PATH="~/anaconda3/bin:~/Library/TeX/texbin:$PATH"`

If you are using Ubuntu or an older version of Mac it might be in `/usr/bin` instead. Also, for anyone reading this who isn't familiar with editing bashrc/bash_profile, make sure you run `source ~/.bash_profile` or `source ~\.bashrc` to apply the changes, or simply close and reopen the terminal window you're working in. 

I'm still pretty confused as to how I was able to create the plot above without making this change / why importing Eureka produced the error. However, I'm not sure it's worth spending too much more time figuring things out. If you're comfortable with this current solution Kevin feel free to close this issue.
